lib/config/migrations/custom/unpublish-safe-migration.ts

```diff
@@ -0,0 +1,43 @@
+import is from '@sindresorhus/is';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class UnpublishSafeMigration extends AbstractMigration {
+  private static readonly SUPPORTED_VALUES = [
+    ':unpublishSafe',
+    'default:unpublishSafe',
+    'npm:unpublishSafe',
+  ];
+
+  override readonly deprecated = true;
+  override readonly propertyName = 'unpublishSafe';
+
+  override run(value: unknown): void {
+    const extendsValue = this.get('extends');
+    const newExtendsValue = Array.isArray(extendsValue) ? extendsValue : [];
+
+    if (value === true) {
+      if (is.string(extendsValue)) {
+        newExtendsValue.push(extendsValue);
+      }
+
+      if (newExtendsValue.every((item) => !this.isSupportedValue(item))) {
+        newExtendsValue.push('npm:unpublishSafe');
+      }
+
+      this.setHard(
+        'extends',
+        newExtendsValue.map((item) => {
```

Comment:
This logic looks different than original and can potentially cause duplicate presets.

Reply:
It should not, because I only added renaming

Reply:
This is different, as it can now cause multiple `npm:unpublishSafe` values, because of the `map` function. The original code uses the `some` function and only add if not yet added.

Reply:
Originally if you have, for example, `{ unpublishSafe: [':unpublishSafe'] }`, we just migrate it to `{ extends: [':unpublishSafe'] }`. Then, migration script handles `extends` property and migrate it to `{ extends: ['npm:unpublishSafe'] }`. I decided to do it one step, otherwise not possible to write correct unit test for specific migration.

Reply:
sure, but you need to dedplicate, any of the `` :unpublishSafe` presets can already be in the extends, so we don't want have them twice

Reply:
I have added unit test
```
it('prevent duplicates', () => {
    expect(UnpublishSafeMigration).toMigrate(
      {
        extends: ['npm:unpublishSafe'],
        unpublishSafe: true,
      },
      {
        extends: ['npm:unpublishSafe'],
      }
    );
  });
```
So, I don't see option to make duplicates. It possible only if `extends` already contains duplicates